link/perf_event.go

```diff
@@ -69,6 +73,12 @@ type perfEvent struct {
 	// True for kretprobes/uretprobes.
 	ret bool
 
+	// Indicates whether this is an uprobe or a different type of perf event.
+	// Needed when cleaning up tracefs events.
+	//
+	// TODO(matt): store a perfEventType in here?
```

Comment:
What types would there be? `kprobe`, `uprobe`, `tracepoint`?

Reply:
Yes

Reply:
Lack of foresight on my part, I didn't realize `Uprobe` wasn't a discrete `ProgramType`. Both uprobes and kprobes attach `Kprobe` programs.

A `type perfEventType` field here could replace `ret`, `isUprobe` and `progType` fields. `perfEventType` would be an enum of `tracepoint`, `kprobe`, `kretprobe`, `uprobe`, `uretprobe`. The switch on `pe.progType` in `(*perfEvent).Close()` would become a switch on `pe.type`.

Reply:
https://github.com/cilium/ebpf/pull/283

Reply:
Perfect :ok_hand: